chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/PendingSurveillanceManager.java

```diff
@@ -1,31 +1,685 @@
+    private UserPermissionDAO userPermissionDAO;
+    private CertifiedProductDAO cpDAO;
 
-    void rejectPendingSurveillance(Long id)
-            throws ObjectMissingValidationException, JsonProcessingException, EntityRetrievalException,
-            EntityCreationException;
+    @Autowired
+    public PendingSurveillanceManager(Environment env, FileUtils fileUtils,
+            SurveillanceUploadManager survUploadManager, JobManager jobManager,
+            UserManager userManager, CertifiedProductManager cpManager, SurveillanceDAO survDao, UserDAO userDao,
+            ActivityManager activityManager, CertifiedProductDetailsManager cpDetailsManager,
+            SurveillanceCreationValidator survCreationValidator,
+            @Qualifier("surveillanceUpdateValidator") SurveillanceUpdateValidator survUpdateValidator,
+            UserPermissionDAO userPermissionDAO, CertifiedProductDAO cpDAO) {
+        this.env = env;
+        this.fileUtils = fileUtils;
+        this.survUploadHelper = survUploadManager;
+        this.jobManager = jobManager;
+        this.userManager = userManager;
+        this.cpManager = cpManager;
+        this.survDao = survDao;
+        this.userDao = userDao;
+        this.activityManager = activityManager;
+        this.cpDetailsManager = cpDetailsManager;
+        this.survCreationValidator = survCreationValidator;
+        this.survUpdateValidator = survUpdateValidator;
+        this.userPermissionDAO = userPermissionDAO;
+        this.cpDAO = cpDAO;
+    }
+
+    @Transactional
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).PENDING_SURVEILLANCE, "
+            + "T(gov.healthit.chpl.permissions.domains.PendingSurveillanceDomainPermissions).UPLOAD)")
+    public SurveillanceUploadResult uploadPendingSurveillance(final MultipartFile file)
+            throws ValidationException, EntityCreationException, EntityRetrievalException {
+        if (file.isEmpty()) {
+            throw new ValidationException("You cannot upload an empty file!");
+        }
+
+        if (!file.getContentType().equalsIgnoreCase("text/csv")
+                && !file.getContentType().equalsIgnoreCase("application/vnd.ms-excel")) {
+            throw new ValidationException("File must be a CSV document.");
+        }
+
+        Integer surveillanceThresholdToProcessAsJob = getSurveillanceRecordThreshold();
+
+        // first we need to count how many surveillance records are in the file
+        // to know if we handle it normally or as a background job
+        String data = fileUtils.readFileAsString(file);
+
+        // This is a container used for 2 different result types...
+        SurveillanceUploadResult uploadResult = new SurveillanceUploadResult();
+
+        int numSurveillance = survUploadHelper.countSurveillanceRecords(data);
+        if (numSurveillance < surveillanceThresholdToProcessAsJob) {
+            uploadResult = processAsFile(file);
+        } else { // process as job
```

Comment:
Normally I'm not the type to advocate against comments, but in this case, the comment is the same as the method name so I think it should probably be removed
// process as job == processUploadAsJob(data);